src/document/DocumentAnnotator.ts

```diff
@@ -106,10 +111,12 @@ export default class DocumentAnnotator extends BaseAnnotator {
             return;
         }
 
-        if (mode === Mode.HIGHLIGHT) {
-            this.annotatedEl.classList.add('ba-is-highlighting');
-        } else {
-            this.annotatedEl.classList.remove('ba-is-highlighting');
+        this.removeAnnotationClasses();
+
+        const annotatedElement = this.annotatedEl;
+        const className = ANNOTATION_CLASSES[mode];
+        if (className) {
```

Comment:
Does the `classList.add` choke in some browsers if `undefined` is passed into it?

Reply:
That wasn't my original concern - I was more so thinking that if for some reason className was `undefined`, we probably don't want to be adding that to the classList just because classList should contain valid classes and `undefined` isn't really valid.